### PR TITLE
Adding schema validation for config.yml

### PR DESCRIPTION
**Pull request by SamWheating**
Re: https://github.com/apache/airflow/issues/12357

Adding a schema for config.yml and a schema validation test to the pre-commit hook. I also had to fix up this file in a few spots where the schema validation exposed errors.

Tested this by adding invalid data to config.yml and trying to commit. 

---

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/config_templates/config.yml.schema.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            }
+          },
+          "required": [
+            "name",
+            "description",
+            "version_added",
+            "type",
+            "example",
+            "default"
+          ]
```
Comment (ashb) [inline airflow/config_templates/config.yml.schema.json]:
```suggestion
          ],
          "additional_properties": false
```

File: airflow/config_templates/config.yml.schema.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        }
+      }
+    },
+    "required": [
+      "name",
+      "description",
+      "options"
+    ]
```
Comment (ashb) [inline airflow/config_templates/config.yml.schema.json]:
```suggestion
    ],
    "additional_properties": false
```

File: airflow/config_templates/config.yml.schema.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+              "type": ["string", "null"]
+            },
+            "type": {
+              "type": "string"
+            },
+            "example": {
+              "type": ["string", "null"]
+            },
+            "default": {
+              "type": ["string", "null"]
```
Comment (ashb) [inline airflow/config_templates/config.yml.schema.json]: This should allow numbers too I think

File: airflow/config_templates/config.yml.schema.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+              "type": "string"
+            },
+            "description": {
+              "type": ["string", "null"]
+            },
+            "version_added": {
+              "type": ["string", "null"]
+            },
+            "type": {
+              "type": "string"
```
Comment (ashb) [inline airflow/config_templates/config.yml.schema.json]: Might be worth making this an enum type?

**Review by ashb — CHANGES_REQUESTED** [review]
Nice!

File: airflow/config_templates/config.yml.schema.json (hunk 3 above)
Comment (SamWheating) [reply airflow/config_templates/config.yml.schema.json]: Yeah you're right, I think `example` probably should as well.

**Review by kaxil — APPROVED** [review]

**Review by potiuk — APPROVED** [review]
Cool!

File: airflow/config_templates/config.yml.schema.json [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+              "type": "string",
+              "enum": ["string", "boolean", "integer", "float"]
+            },
+            "example": {
+              "type": ["string", "null", "number"]
+            },
+            "default": {
+              "type": ["string", "null", "number"]
+            },
+            "sensitive": {
```
Comment (mik-laj) [inline airflow/config_templates/config.yml.schema.json:37]: Can you add description to this field?  It is not obvious when this field should be set.

File: airflow/config_templates/config.yml.schema.json [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+              "type": "boolean",
+              "description": "'true' is this value is sensitive and can be specified using {section}___{name}__SECRET or {section}___{name}__CMD environment variables."
```
Comment (mik-laj) [inline airflow/config_templates/config.yml.schema.json]:
```suggestion
              "description": "When true, this option is sensitive and can be specified using AIRFLOW__{section}___{name}__SECRET or AIRFLOW__{section}___{name}__CMD environment variables. See: airflow.configuration.AirflowConfigParser.sensitive_config_values"
```

**Comment by mik-laj** [comment]
@ashb Would you like to add anything?

**Review by ashb — APPROVED** [review]

**Pull request merged**